StardewArchipelago/Locations/CodeInjections/Modded/MagicModInjections.cs

```diff
@@ -107,36 +114,51 @@ public static bool OnCast_AnalyzeGivesLocations_Prefix(object __instance, Farmer
 
         }
 
-        public static void Update_ReplaceMarlonRecipes_Postfix(ShopMenu __instance, GameTime time)
+
+        private static void ReplaceCraftsanityRecipes(ShopMenu shopMenu, Hint[] myActiveHints)
         {
-            try
+            if (!_archipelago.SlotData.Craftsanity.HasFlag(Craftsanity.All))
             {
-                // We only run this once for each menu
-                if (_lastShopMenuUpdated == __instance || __instance.storeContext != "AdventureGuild")
+                return;
+            }
+
+            foreach (var (shopIdentification, recipes) in craftsanityRecipes)
+            {
+                if (!shopIdentification.IsCorrectShop(shopMenu))
                 {
-                    return;
+                    continue;
+                }
+                foreach (var recipe in recipes)
+                {
+                    _shopReplacer.PlaceShopRecipeCheck(shopMenu.itemPriceAndStock, $"{recipe.ItemName} Recipe", recipe.ItemName, myActiveHints, recipe.Price);
                 }
+            }
+        }
 
-                _lastShopMenuUpdated = __instance;
-                if (!_archipelago.SlotData.Craftsanity.HasFlag(Craftsanity.All))
+
+        public static void Update_ReplaceMarlonShopChecks_Postfix(ShopMenu __instance, GameTime time)
```

Comment:
Please add the signature of the function you're patching above the patch like in other places